### PR TITLE
doc: provide better build instructions in debug mode

### PR DESCRIPTION
**Pull request by gxyd**
## Description

Provide better build instructions in debug mode, following a few repeated questions here:
* https://lfortran.zulipchat.com/#narrow/channel/197339-General/topic/Issue.20.235292
* https://lfortran.zulipchat.com/#narrow/channel/197339-General/topic/issue.20.235604

I haven't made any changes to documenting building in *release mode* to guide towards using `ninja` instead of using `make`, as I'm not really sure if we want to do that or not.

### CODE REVIEW

**Comment by assem2002** [comment]
It's very helpful to keep the documentation updated. One thing a developer hates is unclear installation instructions :).

File: doc/src/installation.md [hunk 1]
```diff
@@ -90,10 +90,19 @@ Generate files that are needed for the build (this step depends on `re2c`, `biso
...
 ```
-cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_LLVM=yes -DCMAKE_INSTALL_PREFIX=`pwd`/inst .
-make -j8
+
+and can use `ninja` to rebuild.
+
+To do a clean rebuild, you can use:
+```bash
+# NOTE: the below git command deletes all untracked files
+git clean -dfx  # reset repository to a clean state by removing artifacts generated during the build process
```
Comment (Pranavchiku) [inline doc/src/installation.md:103]: > reset repository to a clean state by removing artifacts generated during the build process

This gives a intuition that build will generate files, but it does not, so I would rephrase it

Comment (gxyd) [reply doc/src/installation.md:103]: when I do `git clean -dfx`, I get the below as output:
```console
Removing .ninja_deps
Removing .ninja_log
Removing CMakeCache.txt
Removing CMakeFiles/
Removing CPackConfig.cmake
Removing CPackSourceConfig.cmake
Removing CTestTestfile.cmake
Removing a.out
Removing build.ninja
Removing cmake/postinstall/CMakeFiles/
Removing cmake/postinstall/CTestTestfile.cmake
Removing cmake/postinstall/cmake_install.cmake
Removing cmake_install.cmake
Removing compile_commands.json
Removing doc/man/CMakeFiles/
Removing doc/man/CTestTestfile.cmake
Removing doc/man/cmake_install.cmake
Removing inst/
Removing install_manifest.txt
Removing integration_tests/intrinsics_331.tmp.o
Removing integration_tests/test-llvm/
Removing intrinsics_331.out
Removing pass_00_initial_asr_01.clj
Removing pass_00_initial_asr_02.clj
Removing pass_01_nested_vars.clj
Removing pass_02_global_stmts.clj
Removing pass_03_transform_optional_argument_functions.clj
Removing pass_04_init_expr.clj
Removing pass_05_forall.clj
Removing pass_06_openmp.clj
Removing pass_07_implied_do_loops.clj
Removing pass_08_class_constructor.clj
Removing pass_09_pass_list_expr.clj
Removing pass_10_where.clj
Removing pass_11_function_call_in_declaration.clj
Removing pass_12_subroutine_from_function.clj
Removing pass_13_array_op.clj
Removing pass_14_symbolic.clj
Removing pass_15_intrinsic_function.clj
Removing pass_16_intrinsic_subroutine.clj
Removing pass_17_subroutine_from_function.clj
Removing pass_18_array_op.clj
Removing pass_19_pass_array_by_data.clj
Removing pass_20_print_struct_type.clj
Removing pass_21_print_arr.clj
Removing pass_22_print_list_tuple.clj
Removing pass_23_print_struct_type.clj
Removing pass_24_array_dim_intrinsics_update.clj
Removing pass_25_do_loops.clj
Removing pass_26_while_else.clj
Removing pass_27_select_case.clj
Removing pass_28_inline_function_calls.clj
Removing pass_29_unused_functions.clj
Removing pass_30_unique_symbols.clj
Removing pass_31_insert_deallocate.clj
Removing src/CMakeFiles/
Removing src/CTestTestfile.cmake
Removing src/bin/CMakeFiles/
Removing src/bin/CTestTestfile.cmake
Removing src/bin/cmake_install.cmake
Removing src/bin/lfortran
Removing src/bin/lfortran.dSYM/
Removing src/cmake_install.cmake
Removing src/lfortran/CMakeFiles/
Removing src/lfortran/CTestTestfile.cmake
Removing src/lfortran/ast.h
Removing src/lfortran/cmake_install.cmake
Removing src/lfortran/config.h
Removing src/lfortran/liblfortran_lib.a
Removing src/lfortran/parser/parser.output
Removing src/lfortran/parser/parser.tab.cc
Removing src/lfortran/parser/parser.tab.hh
Removing src/lfortran/parser/preprocessor.cpp
Removing src/lfortran/parser/tokenizer.cpp
Removing src/lfortran/tests/CMakeFiles/
Removing src/lfortran/tests/CTestTestfile.cmake
Removing src/lfortran/tests/cmake_install.cmake
Removing src/lfortran/tests/test_asr
Removing src/lfortran/tests/test_cwrapper
Removing src/lfortran/tests/test_lfortran
Removing src/lfortran/tests/test_stacktrace
Removing src/libasr/CMakeFiles/
Removing src/libasr/CTestTestfile.cmake
Removing src/libasr/__pycache__/
Removing src/libasr/asr.h
Removing src/libasr/asr_base_visitor.h
Removing src/libasr/asr_deserialization_visitor.h
Removing src/libasr/asr_expr_base_replacer_visitor.h
Removing src/libasr/asr_expr_call_replacer_visitor.h
Removing src/libasr/asr_expr_stmt_duplicator_visitor.h
Removing src/libasr/asr_expr_type_visitor.h
Removing src/libasr/asr_expr_value_visitor.h
Removing src/libasr/asr_json_visitor.h
Removing src/libasr/asr_lookup_name_visitor.h
Removing src/libasr/asr_pass_walk_visitor.h
Removing src/libasr/asr_pickle_visitor.h
Removing src/libasr/asr_serialization_visitor.h
Removing src/libasr/asr_stmt_base_replacer_visitor.h
Removing src/libasr/asr_tree_visitor.h
Removing src/libasr/asr_walk_visitor.h
Removing src/libasr/cmake_install.cmake
Removing src/libasr/config.h
Removing src/libasr/libasr.a
Removing src/libasr/pass/intrinsic_function_registry_util.h
Removing src/libasr/wasm_visitor.h
Removing src/runtime/.ninja_log
Removing src/runtime/CMakeCache.txt
Removing src/runtime/CMakeFiles/
Removing src/runtime/build.ninja
Removing src/runtime/cmake_install.cmake
Removing src/runtime/install_manifest.txt
Removing src/runtime/legacy/CMakeFiles/
Removing src/runtime/legacy/CTestTestfile.cmake
Removing src/runtime/legacy/cmake_install.cmake
Removing src/runtime/legacy/liblfortran_runtime_static.a
Removing src/runtime/lfortran_intrinsic_custom.mod
Removing src/runtime/lfortran_intrinsic_ieee_arithmetic.mod
Removing src/runtime/lfortran_intrinsic_iso_c_binding.mod
Removing src/runtime/lfortran_intrinsic_iso_fortran_env.mod
Removing src/runtime/liblfortran_runtime.0.42.0.dylib
Removing src/runtime/liblfortran_runtime.0.dylib
Removing src/runtime/liblfortran_runtime.dylib
Removing src/runtime/omp_lib.mod
Removing src/tests/CMakeFiles/
Removing src/tests/CTestTestfile.cmake
Removing src/tests/cmake_install.cmake
Removing src/tests/libdoctest.a
Removing test.out
Removing test.tmp.o
Removing version
```

So, it does seems like it removes the files *generated* during build.

**Review by certik — APPROVED** [review]
